lib/src/ardrive_http.dart

```diff
@@ -87,6 +92,61 @@ class ArDriveHTTP {
     return get(url: url, responseType: ResponseType.bytes);
   }
 
+  Future<ArDriveHTTPResponse> getAsByteStream(String url, {Completer<String>? cancelWithReason}) async {
+    return kIsWeb 
+      ? _getAsByteStreamWeb(url, cancelWithReason: cancelWithReason)
+      : _getAsByteStreamIO(url, cancelWithReason: cancelWithReason);
+  }
+
+  Future<ArDriveHTTPResponse> _getAsByteStreamWeb(String url, {Completer<String>? cancelWithReason}) async {
+    final client = fetch.FetchClient(mode: fetch.RequestMode.cors);
+    
+    final response = await client.send(
+      http.Request(
```

Comment:
we don't `catch` errors. We could just throw an `ArDriveHTTPException` when get errors.